examples/container/inspect/src/main.rs

```diff
@@ -0,0 +1,100 @@
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+use caps::CapSet;
+use nix::{libc::c_int, unistd};
+use signal_hook::{consts::signal::*, low_level};
+use std::{env, fs};
+
+type Signals =
+    signal_hook::iterator::SignalsInfo<signal_hook::iterator::exfiltrator::origin::WithOrigin>;
+
+const SIGNALS: &[c_int] = &[
+    SIGTERM, SIGQUIT, SIGINT, SIGTSTP, SIGWINCH, SIGHUP, SIGCHLD, SIGCONT,
+];
+
+fn dump(file: &str) {
+    println!("{}:", file);
+    fs::read_to_string(file)
+        .unwrap_or_else(|_| panic!("dump {}", file))
+        .lines()
+        .for_each(|l| println!("  {}", l));
+}
+
+fn main() {
+    env::set_var("RUST_BACKTRACE", "1");
+
+    println!("getpid: {}", unistd::getpid());
+    println!("getppid: {}", unistd::getppid());
+    println!("getuid: {}", unistd::getuid());
+    println!("getgid: {}", unistd::getgid());
+    let pwd = env::current_dir().expect("current_dir");
+    println!("pwd: {}", pwd.display());
+    let exe = env::current_exe().expect("current_exe");
+    println!("exe: {}", exe.display());
+
+    dump("/proc/self/mounts");
+    dump("/proc/self/environ");
+
+    println!("fds:");
+    for entry in fs::read_dir("/proc/self/fd").expect("read_dir /proc/self/fd") {
+        let entry = entry.unwrap().path();
+        let link = fs::read_link(&entry).expect("mount entry");
+        println!("    {}: {}", entry.display(), link.display());
+    }
+
+    for set in &[
+        CapSet::Ambient,
+        CapSet::Bounding,
+        CapSet::Effective,
+        CapSet::Inheritable,
+        CapSet::Permitted,
+    ] {
+        println!(
+            "caps {}: {:?}",
+            format!("{:?}", set).as_str().to_lowercase(),
+            caps::read(None, *set).expect("Failed to read caps")
+        );
+    }
+
+    println!("ps:");
+    for entry in fs::read_dir("/proc").expect("read_dir /proc") {
+        let entry = entry.unwrap().path();
+        if !entry.is_dir() {
+            continue;
+        }
+        let pid = entry.file_name().unwrap().to_str().unwrap();
+        let pid = match pid.parse::<u32>() {
+            Ok(pid) => pid,
+            Err(_) => continue,
+        };
+
+        let stat = std::fs::read_to_string(entry.join("stat")).expect("read stat");
+        let cmdline = std::fs::read_to_string(entry.join("cmdline")).expect("read cmdline");
+        let name = stat
+            .split_whitespace()
+            .nth(1)
+            .unwrap()
+            .trim_start_matches('(')
+            .trim_end_matches(')');
+        println!("    {:>10}: {:<10}: {}", pid, name, cmdline);
+    }
+
+    let mut sigs = Signals::new(SIGNALS).expect("install signal handler");
+    for signal in &mut sigs {
+        eprintln!("Received signal {:?}", signal);
+        let signal = signal.signal;
+        low_level::emulate_default_handler(signal).expect("default signal handler");
```

Comment:
even after reading the documentation I'm not sure what this is doing

Reply:
This handler prints the received signal and does best effort to do the same as the default handler would do. Probably just exit.